xds/src/main/java/io/grpc/xds/client/XdsClientImpl.java

```diff
@@ -738,6 +740,9 @@ void restartTimer() {
         // When client becomes ready, it triggers a restartTimer for all relevant subscribers.
         return;
       }
+      ServerInfo serverInfo = activeCpc.getServerInfo();
+      int timeoutSec = xdsDataErrorHandlingEnabled && serverInfo.resourceTimerIsTransientError()
```

Comment:
Checking `xdsDataErrorHandlingEnabled` doesn't seem to do anything, as the flag protecting is happening in BootstrapperImpl.